ck_ssl.c

```diff
@@ -1529,82 +1529,6 @@ ssl_once_init()
     debug(F110,"OpenSSL Library",SSLeay_version(SSLEAY_CFLAGS),0);
     debug(F110,"OpenSSL Library",SSLeay_version(SSLEAY_PLATFORM),0);
 
-    /* The following test is suggested by Richard Levitte */
-    /* if (((OPENSSL_VERSION_NUMBER ^ SSLeay()) & 0xffffff0f) */
-    /* Modified by Adam Friedlander for OpenSSL >= 1.0.0 */
-    if (OPENSSL_VERSION_NUMBER > SSLeay()
-         || ((OPENSSL_VERSION_NUMBER ^ SSLeay()) & COMPAT_VERSION_MASK)
-#ifdef OS2
-/* DG 2024-08-05: Not sure what the point of this was. Presumably the goal was
```

Comment:
@davidrg The problem on both OS2 and Windows is that OpenSSL DLLs are built and distributed as part of many applications.   Whichever one is found in the PATH first will be loaded unless there is some binding between the application and the associated DLLs.   Windows introduced the concept of an assembly to work around this problem.

If the wrong version of OpenSSL DLLs were loaded, then the process could crash.

Reply:
The block of comments in ck_ssl.c from line 1500 explain when and why the checks are required.
If the ABI guarantees have changed for major version 3, then the comment block should be updated to explain the changes and the version check should be modified to accept the new guarantees when the major version is 3.